Detectors/MUON/MCH/Base/include/MCHBase/Digit.h

```diff
@@ -28,34 +28,20 @@ namespace mch
 class Digit
 {
  public:
-  struct Time {
-    union {
-      // default value
-      uint64_t time = 0x0000000000000000;
-      struct {                       ///
-        uint32_t sampaTime : 10;     /// bit 0 to 9: sampa time
-        uint32_t bunchCrossing : 20; /// bit 10 to 29: bunch crossing counter
-        uint32_t reserved : 2;       /// bit 30 to 31: reserved
-        uint32_t orbit;              /// bit 32 to 63: orbit
-      };                             ///
-    };
-    uint64_t getBXTime()
-    {
-      return (bunchCrossing + (sampaTime * 4));
-    }
-  };
-
   Digit() = default;
 
-  Digit(int detid, int pad, unsigned long adc, Time time, uint16_t nSamples = 1);
+  Digit(int detid, int pad, unsigned long adc, int32_t time, uint16_t nSamples = 1);
   ~Digit() = default;
 
   bool operator==(const Digit&) const;
 
-  Time getTime() const { return mTime; }
+  // time in bunch crossing units, relative to the beginning of the TimeFrame
+  void setTime(int32_t t) { mTFtime = t; }
+  int32_t getTime() const { return mTFtime; }
 
-  uint16_t nofSamples() const { return mNofSamples; }
   void setNofSamples(uint16_t n) { mNofSamples = n; }
```

Comment:
We probably need 2 separate setters too, using masks to avoid interferences

Reply:
Added, but the constructor still takes only the number of samples parameter. Now the constructor sets the "saturated" flag to false, and it needs to be explicitly set afterwards.